WebRoot/js/zimbraMail/share/view/dialog/ZmPasswordRecoveryDialog.js

```diff
@@ -85,31 +91,48 @@ ZmPasswordRecoveryDialog.REQUEST_CODE_BUTTON = ++DwtDialog.LAST_BUTTON;
 ZmPasswordRecoveryDialog.RESEND_OPTION_BUTTON = ++DwtDialog.LAST_BUTTON;
 ZmPasswordRecoveryDialog.VERIFY_CODE_BUTTON = ++DwtDialog.LAST_BUTTON;
 ZmPasswordRecoveryDialog.RESET_SUBMIT_BUTTON = ++DwtDialog.LAST_BUTTON;
+ZmPasswordRecoveryDialog.LOGIN_BUTTON = ++DwtDialog.LAST_BUTTON;
 
-ZmPasswordRecoveryDialog.prototype.toString = function() {
-	return "ZmPasswordRecoveryDialog";
+/**
+ * Returns the strng name of this class.
+ *
+ * @return class name
+ */
+ZmPasswordRecoveryDialog.prototype.toString =
+function() {
+	return 'ZmPasswordRecoveryDialog';
 };
 
 /**
  * Gets the HTML that forms the basic framework of the dialog.
  *
- * @private
  */
-ZmPasswordRecoveryDialog.prototype._contentHtml = function() {
+ZmPasswordRecoveryDialog.prototype._contentHtml =
+function() {
 	var id = this._htmlElId;
-	this._getRecoveryAccountDivId = id + "_get_recovery_account";
-	this._requestCodeDivId = id + "_request_code";
-	this._validateCodeDivId = id + "_validate_code";
-	this._validateCodeErrorDivId = id + "_validate_code_error";
-	this._codeSuccessDivId = id + "_code_success";
-	this._resetPasswordDivId = id + "_reset_password";
-	this._resetPasswordDescriptionDivId = id + "_reset_password_description";
-	this._resetPasswordErrorDivId = id + "_reset_password_error";
-	this._passwordResetSuccessDivId = id + "_password_reset_success";
+	this._getRecoveryAccountDivId = id + '_get_recovery_account';
+	this._requestCodeDivId = id + '_request_code';
+	this._validateCodeDivId = id + '_validate_code';
+	this._codeSuccessDivId = id + '_code_success';
+	this._resetPasswordDivId = id + '_reset_password';
+	this._resetPasswordDescriptionDivId = id + '_reset_password_description';
+	this._passwordResetSuccessDivId = id + '_password_reset_success';
 	this._divIdArray = [this._getRecoveryAccountDivId, this._requestCodeDivId, this._validateCodeDivId, this._codeSuccessDivId, this._resetPasswordDivId, this._passwordResetSuccessDivId];
-	return AjxTemplate.expand("share.Dialogs#ZmPasswordRecovery", {id : id, accountInput : this.accountInput});
+	return AjxTemplate.expand('share.Dialogs#ZmPasswordRecovery', {id : id, accountInput : this.accountInput});
 };
 
+/**
+ * Create Recovery Button method.
+ *
+ * To reduce repetitive code, make a function that can build password recovery buttons with the desired params.
```

Comment:
Is this comment what you needed to do before writing the method? I dislike this description.

Reply:
Updated in commit 20098c4.
 * Create DWT buttons that are not part of the main dialog button flow.
 * These buttons are to be used in the body of the dialog view.